bravado/client.py

```diff
@@ -239,27 +239,20 @@ def __call__(self, **op_kwargs):
         log.debug(u'%s(%s)', self.operation.operation_id, op_kwargs)
         warn_for_deprecated_op(self.operation)
 
-        # Apply request_options defaults
-        request_options = dict(
-            REQUEST_OPTIONS_DEFAULTS,
-            **(op_kwargs.pop('_request_options', {})))
+        # Get per-request config
+        request_options = op_kwargs.pop('_request_options', {})
+        request_config = RequestConfig.from_request_options_dict(request_options, self.also_return_response)
```

Comment:
I have some doubts about setting a named tuple to keep request config data.
At the moment bravado internally needs/uses few keys so that seems reasonable, but there are few keys (ie. `tcp_nodelay` used by fido) that are actually defined as HttpClient specific.
Using a namedtuple will prevent exchange of request options that are strictly dependent on HTTP client and/or we'll have to expand the namedtuple attributes "everytime" it could happen.

Even if that was not an intended feature, I think that it would be cool to keep.
What about converting the named tuple that keeps argument in input and allows and stores kwargs too?
In this way to could have _stricter_ control on the request configuration passed around and give free space to any further extension

As this change have the potential to reduce extendibility I would like to iterate a bit more on this.

Reply:
I agree it could prove to be problematic to be that strict; let's discuss offline. One thing I'll mention is that `tcp_nodelay` is _not_ passed through as a request_option, so is unrelated to this change. This is why the test for it continues to pass. ;)